### PR TITLE
Code Signing Policy Modification [T1553.006]

### LINKED ISSUE #359: Subvert Trust Controls: Code Signing Policy Modification
https://attack.mitre.org/techniques/T1553/006

### PR DESCRIPTION
**Pull request by williballenthin**
closes #359

### CODE REVIEW

File: host-interaction/bootloader/disable-code-signing.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    author: william.ballenthin@fireeye.com
+    scope: function
+    att&ck:
+      - Defense Evasion::Subvert Trust Controls::Code Signing Policy Modification [T1553.006]
+    examples:
+      - 0596C4EA5AA8DEF47F22C85D75AACA95:0x10710B3  # old Necurs rootkit
+  features:
+    - and:
+      - match: create process
+      - string: /^bcdedit(\.exe)? -set TESTSIGNING ON/i
```
Comment (williballenthin) [inline host-interaction/bootloader/disable-code-signing.yml:14]: this is more commonly seen in BAT scripts (possibly dropped by malware) but, even then, there aren't too many examples of this technique within a program

File: host-interaction/bootloader/manipulate-boot-configuration.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          - string: "/deletevalue safeboot"
+      - and:
+        - string: /boot.ini/i
+        - optional:
+          - api: kernel32.GetPrivateProfileStringA
+          - api: kernel32.WritePrivateProfileString
+          - or:
+            - string: "default"
+            - string: "boot loader"
+            - string: "operating systems"
```
Comment (williballenthin) [inline host-interaction/bootloader/manipulate-boot-configuration.yml:25]: these were just some random things i encountered in the example binary. i can remove/tweak if this rule isn't specific enough

File: host-interaction/driver/disable-driver-code-integrity.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    scope: function
+    references:
+      - https://www.fuzzysecurity.com/tutorials/28.html
+      - https://j00ru.vexillium.org/2010/06/insight-into-the-driver-signature-enforcement/
+    examples:
+      - 31CEE4F66CF3B537E3D2D37A71F339F4:0x140004070
+  features:
+    - or:
+      - string: "CiInitialize"
+        description: exported symbol name used to resolve code integrity configuration
```
Comment (williballenthin) [inline host-interaction/driver/disable-driver-code-integrity.yml:15]: public techniques use memory scanning near this symbol to find the kernel address of the flag and flip it. i'd hoped to use some `offset` features but there's not too much to work with.

most public techniques abuse vulnerable drivers to r/w kernel memory from user code.

File: persistence/registry/appinitdlls/disable-appinit_dlls-code-signature-enforcement.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      - Defense Evasion::Subvert Trust Controls::Code Signing Policy Modification [T1553.006]
+    references:
+      - https://docs.microsoft.com/en-us/windows/win32/win7appqual/appinit-dlls-in-windows-7-and-windows-server-2008-r2
+    examples:
+      - 58ADC2E97FBEE01B71073CCD7FF1B9A4:0x401350
+  features:
+    - and:
+      - string: /RequireSignedAppInit_DLLs/i
+        description: disable DLL code signature enforcement
+      - number: 0 = state disabled
```
Comment (williballenthin) [inline persistence/registry/appinitdlls/disable-appinit_dlls-code-signature-enforcement.yml:18]: this is loose, but i suspect simply the presence of the above value name is likely to be sufficient.

File: host-interaction/bootloader/manipulate-boot-configuration.yml (hunk 2 above)
Comment (mr-tz) [reply host-interaction/bootloader/manipulate-boot-configuration.yml:25]: do we need to consider lower/upper case?

**Review by mr-tz — APPROVED** [review]
LGTM!

File: host-interaction/bootloader/manipulate-boot-configuration.yml (hunk 2 above)
Comment (williballenthin) [reply host-interaction/bootloader/manipulate-boot-configuration.yml:25]: good catch

image

**Pull request merged**